src/org/rascalmpl/util/maven/Util.java

```diff
@@ -41,29 +41,49 @@
 import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
 
 public class Util {
-    public static Path mavenRepository() {
+    public static Path mavenRepository(Settings settings) {
         String repoProp = System.getProperty("maven.repo.local");
         if (repoProp != null) {
             return Path.of(repoProp);
         }
 
+        if (settings.getLocalRepository() != null) {
+            return Path.of(replaceVariables(settings.getLocalRepository()));
+        }
+
+        Path userHome = Path.of(System.getProperty("user.home"));
+        return userHome.resolve(".m2").resolve("repository");
+    }
+
+    public static Settings readSettings() {
```

Comment:
I think the behavior has changed.

Given a setting we want to know, we used to do:

1. look for the setting in `.m2/settings.xml` if it's there, use it
2. look for the setting in  `$MAVEN_HOME/settings.xml` if it's there, use that
3. otherwise, use a default.

But now, this code returns a single setting instance, and we don't know where it came from. If we want to go this route (which I'm not completely convinced) we should at least first get the one from maven_home, and then override it with all the settings from the user dir. Not trivial?

Reply:
Thanks for catching this, my approach was indeed flawed. I have now implemented this more thoroughly.